internal/mux/mux.go

```diff
@@ -315,15 +304,15 @@ func (p *playerImpl) playImpl() {
 	p.state = playerPlay
 
 	if !p.eof {
-		buf, free := p.getTmpBuf()
-		defer free()
+		buf := p.pooledBuf()
+		defer bufPool.Put(buf)
```

Comment:
As `p.buf` holds the reference to `buf`, putting the buffer back doesn't seem correct.